docs/devel/luns.md

```diff
@@ -0,0 +1,56 @@
+Direct LUNs migration
+==============
+
+Summary
+-------
+
+VM contains disks. Some of them might be connected as a direct LUN.
+When a user wish to re-use the LUN disks, he needs them attached to the VM.
+The LUN should be used in a single place, or else, it will cause a data corruption.
+The storage where the LUNs reside, needs to be reachable to the VM environment.
+This feature page mostly about import from oVirt.
+
+Design
+------
+
+The requirements for the LUN migration code are (not in priority order)
+
+1. Gather the LUN informnation from the source destination.
+2. To ease on the user - less manual actions, more automated migration.
+3. Save unnecessary risks.
+
+Requirements
+----------------------------------------
+
+1. Destination (OCP cluster) reachability to the ISCSI storage, where the LUN is.
+2. RBAC permission to create `PersistentVolumes` by the controller.
+
+Risks and Mitigations
+----------------------------------------
+
+1. A LUN being used by a VM the user wish to migrate, without reachability to the destination.
+2. A VM that will be used while or after the migration with a direct LUN connected.
+   - A warning should be shown before starting the migration.
+   - (Optional) UI checkbox to the user, before starting the migration.
+3. An oVirt version with unsupported API. The [PR](https://github.com/oVirt/ovirt-engine/pull/559) provides LUN details is required.
+   - We may check for missing LUN details and skip the disk migration.
+   - (Optional) Have a mechanism to check the oVirt version for future work.
+4. Creating a `PersistentVolume` requires the file system type: `ext4` or `xfs` of the LUN, which we don't know.
```

Comment:
we can't use `volumeMode: Block`?

Reply:
Both required, the `volumeMode: Block` is hard-coded for LUN, but we do have `fsType` spec under the `iscsi`. By default it would be `ext4` (when not provided).

Reply:
isnt't that only for volumemount? if it's block it shoulnd't need any fs.

Reply:
i was based on https://docs.openshift.com/container-platform/4.6/storage/persistent_storage/persistent-storage-iscsi.html , but reading carefully https://kubernetes.io/docs/concepts/storage/persistent-volumes/#volume-mode, maybe it is possible to drop.